HeterogeneousCore/SonicTriton/interface/TritonMemResource.h

```diff
@@ -0,0 +1,90 @@
+#ifndef HeterogeneousCore_SonicTriton_TritonMemResource
+#define HeterogeneousCore_SonicTriton_TritonMemResource
+
+#include <string>
+#include <memory>
+
+#include "HeterogeneousCore/SonicTriton/interface/grpc_client_gpu.h"
+
+//forward declaration
+template <typename IO>
+class TritonData;
+struct cudaIpcMemHandle_st;
+typedef cudaIpcMemHandle_st cudaIpcMemHandle_t;
```

Comment:
I was trying to follow the usual recommendation not to include unnecessary headers in `.h` files, but as you observe, it's a bit complicated in this case because typedefs can't be forward declared in the usual way. It's fine with me just to include the whole header if that is the recommendation in cases like this.